ocular/src/chain/client/authz.rs

```diff
@@ -0,0 +1,171 @@
+                    }
+                    .encode_to_vec(),
+                }),
+                expiration: expiration_timestamp,
+            }),
+        };
+
+        // Build tx body.
+        let msg_any = prost_types::Any {
+            type_url: String::from("/cosmos.authz.v1beta1.MsgGrant"),
+            value: msg.encode_to_vec(),
+        };
+
+        let tx_body = tx::Body::new(vec![msg_any], &tx_metadata.memo, tx_metadata.timeout_height);
+
+        self.sign_and_send_msg(
+            granter.public_key,
+            granter.private_key,
+            tx_body,
+            tx_metadata,
+            None,
+            None,
+        )
+        .await
+    }
+
+    // Revoke Authorization
+    // TODO: support other types of authorization revokes other than send messages.
+    pub async fn revoke_send_authorization(
+        &self,
+        granter: Account,
+        grantee: AccountId,
+        tx_metadata: TxMetadata,
+    ) -> Result<Response, TxError> {
+        let msg = MsgRevoke {
+            granter: granter.id.to_string(),
+            grantee: grantee.to_string(),
+            msg_type_url: String::from("/cosmos.bank.v1beta1.MsgSend"),
+        };
+
+        // Build tx body.
+        let msg_any = prost_types::Any {
+            type_url: String::from("/cosmos.authz.v1beta1.MsgRevoke"),
+            value: msg.encode_to_vec(),
+        };
+
+        let tx_body = tx::Body::new(vec![msg_any], &tx_metadata.memo, tx_metadata.timeout_height);
+
+        self.sign_and_send_msg(
+            granter.public_key,
+            granter.private_key,
+            tx_body,
+            tx_metadata,
+            None,
+            None,
+        )
+        .await
+    }
+
+    // Execute a transaction previously authorized by granter
+    pub async fn execute_authorized_tx(
+        &self,
+        grantee: Account,
+        msgs: Vec<::prost_types::Any>,
+        tx_metadata: TxMetadata,
+        fee_payer: Option<AccountId>,
+        fee_granter: Option<AccountId>,
+    ) -> Result<Response, TxError> {
+        let msg = MsgExec {
+            grantee: grantee.id.to_string(),
+            msgs,
+        };
+
+        // Build tx body.
+        let msg_any = prost_types::Any {
+            type_url: String::from("/cosmos.authz.v1beta1.MsgExec"),
+            value: msg.encode_to_vec(),
+        };
+
+        let tx_body = tx::Body::new(vec![msg_any], &tx_metadata.memo, tx_metadata.timeout_height);
+
+        self.sign_and_send_msg(
+            grantee.public_key,
+            grantee.private_key,
+            tx_body,
+            tx_metadata,
+            fee_payer,
+            fee_granter,
+        )
+        .await
+    }
+
+    // Basic fee allowance
+    pub async fn perform_basic_allowance_fee_grant(
+        &self,
+        granter: Account,
+        grantee: AccountId,
+        expiration: Option<prost_types::Timestamp>,
+        // TODO: Standardize below Coin type to common cosmrs coin type once FeeGrants get looped in.
```

Comment:
This todo is due to feegrants not being in cosmrs yet, so the coin type is the one defined by the cosmos-sdk